src/api/Workspace.ts

```diff
@@ -1,24 +1,61 @@
-import { RegisterRequest } from './methods/register';
+import { ServicesMap, ServicesRequest } from './methods/services';
+import { RegisterServiceRequest } from './methods/registerService';
+import { ComponentsMap, ComponentsRequest } from './methods/components';
+import { ConfigRequest } from './methods/config';
 
-interface Workspace {
-  // Load all the services from url
-  // Load the layout and render components inside
+/**
+ * Workspace is the core service of CapsulaHub, it is responsible for :
+ * - Loading services and components that are included in its configuration (START)
+ * - Allowing them to register themselves (REGISTER)
+ * - Letting them communicate together (SERVICES/COMPONENTS)
+ * - Getting their own configuration (CONFIG)
+ */
+export interface Workspace {
+  /**
+   * Loading all services and components included in workspace configuration
+   * Reject in case :
+   * - Workspace already started
+   * - Workspace start already in progress
+   * - An error with importing a service occurred
+   * - An error with importing a component occurred
+   * - An error with registering a component occurred
+   * @param startRequest
+   */
   start(startRequest: StartRequest): Promise<void>;
 
-  // Returns proxy of a registered service
-  // Can be used by any service during run time to get a proxy of another service
-  service(serviceRequest: ServiceRequest): Promise<ServiceResponse>;
+  /**
+   * Getting promises to each service that has been loaded in the workspace
+   * Reject in case :
+   * - Workspace not started yet
+   * @param servicesRequest
+   */
+  services(servicesRequest: ServicesRequest): Promise<ServicesMap>;
 
-  // Register a service or a component in the workspace
-  // Can be used by any service to register itself and become available with Workspace.service method
-  register(registerRequest: RegisterRequest): Promise<void>;
+  /**
+   * Getting promises to each component that has been loaded in the workspace
+   * Reject in case :
+   * - Workspace not started yet
+   * @param componentsRequest
+   */
+  components(componentsRequest: ComponentsRequest): Promise<ComponentsMap>;
 
-  /*
-  // For future / To be defined
-  // Allow persistence of services state
-  persist({ serviceName: string }): Promise<>;
-  state({ serviceName: string }): Observable<ServiceState>;
-  */
+  /**
+   * Register a service in the workspace.
+   * Reject in case :
+   * - Workspace not started yet
+   * - Service specified in request doesn't exist in workspace configuration
+   * - Service specified in request already registered
+   * - Invalid request
+   * @param registerServiceRequest
+   */
+  registerService(registerServiceRequest: RegisterServiceRequest): Promise<void>;
+
+  /**
+   * Get the configuration of a specific service
+   * Reject in case :
+   * - Service specified in request doesn't exist
+   * @param configRequest
+   */
+  config(configRequest: ConfigRequest): Promise<any>;
```

Comment:
Don't we want to provide some patters for services configuration or do we want to expect just **any** value as a config? We have interfaces for ServiceConfig and ComponentConfig

Reply:
I will redefine this better, it's not exactly about ServiceConfig and ComponentConfig here